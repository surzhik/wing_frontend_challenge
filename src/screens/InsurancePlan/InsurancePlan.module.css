.plansHolder {
    margin-top: 5rem;
    display: flex;
}

.planBox {
    width: 33%;
    display: flex;
    flex-direction: column;
    position: relative;
    z-index: 1;

}

.firstPart {
    min-height: 210px;
    display: flex;
    flex-direction: column;
    padding: 20px;
    background: #fff;
}

.firstPart > div  {
    flex: 1 1 auto;
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
}

.firstPart > div:nth-child(2) {
    color: #999999;
    justify-content: center;
}
.firstPart > div:nth-child(3) {
    font-size: 12px;
    color: #206bef;
}
.firstPart > div:nth-child(3) button {
    margin-top: 0.2rem;
    display: block;
    max-width: 112px;
    width: 100%;
    height: 32px;

}

.firstPart h4 {
    font-size: 21px;
}
.secondPart {
    background: #c6c9d3;
    display: flex;
    flex-direction: column;
    min-height: 255px;
    flex: 1 1;
}


.secondPart > div {
    display: flex;
    flex-direction: column;
    color: #49577e;
    font-size: 13px;
    font-weight: bolder;
    padding: 20px;
}
.secondPart  > div:first-child {
    flex: 1 1 auto;
}
.secondPart > div:last-child {
    height: 80px;
    border-top: 1px solid #eaeaeb;
}

.secondPart ul {
    font-size: 14px;
    padding: 0 0 0 20px;
    margin: 1rem 0 0 0;
    list-style: none;
}

.secondPart ul li {
    position: relative;
}
.secondPart ul li:before{
    content: '';
    display: inline-block;
    position: absolute;
    height: 12px;
    width: 16px;
    left: -20px;
    top: 2px;
    background: url("./img/checkmark.svg") no-repeat;
    background-size: contain;

}

.planBox:nth-child(2) {
    width: 36%;
}

.planBox:nth-child(2) {
    box-shadow: 0 0 15px rgba(0, 0, 0, 0.2);
    z-index: 2;
    margin-top: -25px;
    margin-bottom: -5px;
}

.planBox:nth-child(2) .firstPart {
    padding-top: 45px;
    min-height: 240px;
}

.planBox:nth-child(2) .secondPart ul {
    font-size: 15px;
}

.planPrice {
    font-size: 18px;
    color: #206bef;
    font-weight: bold;
}

.star {
    position: absolute;
    top: 0;
    right: 20px;
    width: 34px;
    height: 55px;
    background: url("./img/star.svg") no-repeat;
    background-size: contain;
    z-index: 2;
}

.Back {
    margin-top: 2rem;
}

@media (max-width: 767px) {
    .plansHolder {
      flex-direction: column;
     margin: 0 15px;
    }
    .planBox,
    .planBox:nth-child(2) {
        width: auto;
    }

    .planBox + .planBox {
        margin-top: 2rem;
    }
    .secondPart {
       min-height: auto;
    }
}
