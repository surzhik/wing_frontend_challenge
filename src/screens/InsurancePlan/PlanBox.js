import React from "react";
import * as PropTypes from "prop-types";
import styles from "./InsurancePlan.module.css";

const PlanBox = ({ plan, index, onSelect }) => {

  return (
    <div className={styles.planBox}>
      {index === 1 && <div className={styles.star} />}
      <div className={styles.firstPart}>
        <div>
          <h4>{plan.name}</h4>
        </div>
        <div>
          <div>Starting at</div>
          <div className={styles.planPrice}>${plan.price}</div>
        </div>
        <div>
          <div>Terms, fees and more info</div>
          <button
            className={index === 1 ? "active" : ""}
            onClick={onSelect}
            data-id={plan.id}
          >
            Select
          </button>
        </div>
      </div>
      <div className={styles.secondPart}>
        <div>
          {plan.protect_against && (
            <>
              Protect your device against
              <ul>
                {plan.protect_against.map(item => (
                  <li key={item.replace(/\s/g, "")}>{item}</li>
                ))}
              </ul>
            </>
          )}
        </div>

        <div>
          <div>Repair deductible: ${plan.repair_deductible}</div>
          <div>Replacement deductible: ${plan.replacement_deductible}</div>
        </div>
      </div>
    </div>
  );
};

PlanBox.propTypes = {
  plan: PropTypes.object,
  index: PropTypes.number,
  onSelect: PropTypes.func
};

PlanBox.defaultProps = {
  index: 0,
  onSelect: () => {},
  plan: {
    name: "{name} is empty",
    price: "{price} is empty",
    repair_deductible: "{repair_deductible} is empty",
    replacement_deductible: "{replacement_deductible} is empty",
    protect_against: ["{protect_against} is empty"]
  }
};

export default PlanBox;
