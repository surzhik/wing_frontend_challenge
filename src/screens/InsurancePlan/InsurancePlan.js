import React from "react";
import * as PropTypes from "prop-types";
import * as routes from "app/routes";
import Back from "common/Back";
import SubscriptionSwitcher from "subscriptions/SubscriptionSwitcher";
import PlanBox from "./PlanBox";
import styles from "./InsurancePlan.module.css";

class insurancePlan extends React.Component {
  componentDidMount() {
    this.fetchData();
  }

  fetchData = () => {
    let { fetchInsurancePlans } = this.props;
    fetchInsurancePlans("all");
  };

  doConfirm = event => {
    const { sku, subId, onSelectClick } = this.props;
    event.preventDefault();
    const insPlanId = event.target.getAttribute("data-id");
    onSelectClick(subId, sku, insPlanId);
  };
  render() {
    let {
      subId,
      insurancePlans,
      subscription,
      navigateSubscription
    } = this.props;
    return (
      <div>
        <SubscriptionSwitcher
          {...(subscription === "att_subscription"
            ? { attSubId: subId }
            : { sprintSubId: subId })}
          attRoute={navigateSubscription}
          sprintRoute={routes.sprintSubscription}
        />
        <div className={styles.Back}>
          <Back to={navigateSubscription(subId)} />
        </div>
        <h1>Select an insurance plan</h1>
        <div className={styles.plansHolder}>
          {insurancePlans.map((plan, index) => (
            <PlanBox
              key={plan.id}
              plan={plan}
              index={index}
              onSelect={this.doConfirm}
            />
          ))}
        </div>
      </div>
    );
  }
}

insurancePlan.propTypes = {
  subId: PropTypes.any,
  fetchInsurancePlans: PropTypes.func,
  onSelectClick: PropTypes.func,
  navigateSubscription: PropTypes.func,
  insurancePlans: PropTypes.array,
  sku: PropTypes.string
};

export default insurancePlan;
