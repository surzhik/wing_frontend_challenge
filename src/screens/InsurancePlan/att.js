import { connect } from "react-redux";
import * as routes from "app/routes";
import { fetchValue as fetchInsurancePlans } from "insurance/insurancePlans/actions";
import { getInsurancePlans } from "reducers";
import InsurancePlan from "./InsurancePlan";

const mapStateToProps = state => {
  const insurancePlans = getInsurancePlans(state, ["WEW", "WDPP5P", "WDP3P"]);
  return {
    insurancePlans,
    navigateSubscription: routes.attSubscription,
    subscription: "att_subscription"
  };
};

const mapDispatchToProps = {
  fetchInsurancePlans,
  onSelectClick: routes.attInsuranceConfirm
};

export default connect(mapStateToProps, mapDispatchToProps)(InsurancePlan);
