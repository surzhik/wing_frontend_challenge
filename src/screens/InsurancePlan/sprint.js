import { connect } from "react-redux";
import * as routes from "app/routes";
import { fetchValue as fetchInsurancePlans } from "insurance/insurancePlans/actions";
import { getInsurancePlans } from "reducers";
import InsurancePlan from "./InsurancePlan";

const mapStateToProps = state => {
  const insurancePlans = getInsurancePlans(state, ["WEW", "WDPP5P", "WDP3P"]);
  return { insurancePlans, navigateSubscription: routes.sprintSubscription, subscription: 'subscription' };
};

const mapDispatchToProps = {
  fetchInsurancePlans,
  onSelectClick: routes.sprintInsuranceConfirm
};

export default connect(mapStateToProps, mapDispatchToProps)(InsurancePlan);
