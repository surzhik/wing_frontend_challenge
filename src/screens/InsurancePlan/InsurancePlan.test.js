import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { shallow } from "enzyme";
import InsurancePlan from "./InsurancePlan";
import SubscriptionSwitcher from "subscriptions/SubscriptionSwitcher";
import PlanBox from "./PlanBox";
const mockStore = configureMockStore();

describe("Insurance Plan", () => {
  const props = {
    insurancePlans: [{ id: 1 }, { id: 2 }, { id: 3 }],
    navigateSubscription: jest.fn(),
    subscription: "subscription",
    fetchInsurancePlans: jest.fn(),
    onSelectClick: jest.fn()
  };

  describe("Insurance Plan structure", () => {
    it("have unchanged structure", () => {
      const store = mockStore({});
      const wrapper = shallow(
        <Provider store={store}>
          <InsurancePlan {...props} />
        </Provider>
      ).dive();
      expect(wrapper).toMatchSnapshot();
    });
  });

  const subscription = shallow(<InsurancePlan {...props} />);
  it("should render", () => {
    expect(subscription).toHaveLength(1);
  });

  it("should have selector", () => {
    expect(subscription.find(SubscriptionSwitcher)).toHaveLength(1);
  });
  it("should have 3 plans current props", () => {
    expect(subscription.find(PlanBox)).toHaveLength(3);
  });
});
