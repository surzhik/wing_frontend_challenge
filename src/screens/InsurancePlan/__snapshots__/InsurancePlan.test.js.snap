// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insurance Plan Insurance Plan structure have unchanged structure 1`] = `
<div>
  <Connect(SubscriptionDecorator)
    attRoute={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    sprintRoute={[Function]}
  />
  <div
    className="Back"
  >
    <Back />
  </div>
  <h1>
    Select an insurance plan
  </h1>
  <div
    className="plansHolder"
  >
    <PlanBox
      index={0}
      key="1"
      onSelect={[Function]}
      plan={
        Object {
          "id": 1,
        }
      }
    />
    <PlanBox
      index={1}
      key="2"
      onSelect={[Function]}
      plan={
        Object {
          "id": 2,
        }
      }
    />
    <PlanBox
      index={2}
      key="3"
      onSelect={[Function]}
      plan={
        Object {
          "id": 3,
        }
      }
    />
  </div>
</div>
`;
